src/executor/executor.spec.ts

```diff
@@ -145,5 +147,31 @@ describe("Task Executor", () => {
       expect(r3).toStrictEqual(r1);
       expect(spy).toHaveBeenCalledTimes(1);
     });
+
+    it('it should emit "end" and "ended" events', async () => {
+      const executor = await TaskExecutor.create({ package: "test", startupTimeout: 0, logger, yagnaOptions });
+      let terminated = false;
+      let terminating = false;
+
+      executor.events.on("terminating", () => {
+        expect(terminating).toBe(false);
+        expect(terminated).toBe(false);
+        terminating = true;
+      });
+
+      executor.events.on("terminated", () => {
```

Comment:
Instead of `let` vars you could go with something more elegant:

```ts
const executor = await TaskExecutor.create({ package: "test", startupTimeout: 0, logger, yagnaOptions });

const onTerminated = jest.fn();

executor.events.on("terminated", onTerminated);

await executor.end();
await executor.end();

expect(onTerminated).toHaveBeenCalledTimes(1);
```

Also, the test name does not correspond to the contents and implementation.